scala/scala-impl/src/org/jetbrains/plugins/hydra/HydraRepository.scala

```diff
@@ -0,0 +1,24 @@
+package org.jetbrains.plugins.hydra
+
+import org.jetbrains.plugins.hydra.settings.HydraApplicationSettings
+
+/**
+  * @author Maris Alexandru
+  */
+object HydraRepository {
+
+  private val globalSettings = HydraApplicationSettings.getInstance()
+
+  def getRepositoryUrl: String = {
+    val url = globalSettings.hydraRepositoryUrl
+
+    if(url.endsWith("/")) url.dropRight(1) else url
+  }
+
+  def getRepositoryName: String = {
+    globalSettings.hydraRepositoryUrl.split("/") match {
+      case Array(_, "", repoName, _*) => repoName
```

Comment:
I'm afraid this is going to be very fragile. But before discussing this, we need to see what we do with the credential's realm - see the later comment on this point.